src/envoy/http/service_control/client_cache.cc

```diff
@@ -304,12 +307,25 @@ CancelFunc ClientCache::callCheck(
       request, response,
       [this, response, on_done](const Status& status) {
         CheckResponseInfo response_info;
+
+        // Transportation is successful so parse the checkResponse and use its
+        // status in callback.
         if (status.ok()) {
           Status converted_status = ::espv2::api_proxy::service_control::
               RequestBuilder::ConvertCheckResponse(
                   *response, config_.service_name(), &response_info);
+          ServiceControlFilterStats::collectCheckStatus(
```

Comment:
I don't think this implements what is actually asked for in b/155209585. I believe they want to track each HTTP call to SC Check / Report, not the number of client API calls.

I am working on b/155209585 for what you seem to have implemented here.

Reply:
It is not b/155209585. b/155212712 simply wants to tracks the statuses of all the calls to SC.

Reply:
Sorry I made a typo (edited original comment now). I don't think you implemented the ask in b/155212712.

For example here: https://github.com/GoogleCloudPlatform/esp-v2/pull/151/files#diff-dd87d549727ba1b173020a7adbdc168aR328

This will run for every consumer API call, not for every call to SC Check. If there are 5 API calls and only 1 Check call (because of the cache), you will record the metric 5 times instead of just once.

I may be misunderstanding your comment.

Reply:
Good catch. Done.